docs/concepts/models.md

```diff
@@ -524,182 +500,126 @@ except ValidationError as e:
+        m.a = 'not an int'
 
-    m = Model(a=0)
-    # note: the `model_config` setting validate_assignment=True` can prevent this kind of misbehavior
-    m.a = 'not an int'
-
-    try:
-        m2 = Model.model_validate(m)
-    except ValidationError as e:
-        print(e)
-        """
-        1 validation error for Model
-        a
-          Input should be a valid integer, unable to parse string as an integer [type=int_parsing, input_value='not an int', input_type=str]
-        """
-    ```
+        try:
+            m2 = Model.model_validate(m)
+        except ValidationError as e:
+            print(e)
+            """
+            1 validation error for Model
+            a
+              Input should be a valid integer, unable to parse string as an integer [type=int_parsing, input_value='not an int', input_type=str]
+            """
+        ```
 
 ### Creating models without validation
 
-Pydantic also provides the [`model_construct()`][pydantic.main.BaseModel.model_construct] method, which allows models to be created **without validation**. This
-can be useful in at least a few cases:
+Pydantic also provides the [`model_construct()`][pydantic.main.BaseModel.model_construct] method, which allows models to be created **without validation**. 
+This can be useful in at least a few cases:
 
 * when working with complex data that is already known to be valid (for performance reasons)
-* when one or more of the validator functions are non-idempotent, or
+* when one or more of the validator functions are non-idempotent
 * when one or more of the validator functions have side effects that you don't want to be triggered.
 
 !!! note
-    In Pydantic V2, the performance gap between `BaseModel.__init__` and `BaseModel.model_construct` has been narrowed
-    considerably. For simple models, calling `BaseModel.__init__` may even be faster. If you are using [`model_construct()`][pydantic.main.BaseModel.model_construct]
-    for performance reasons, you may want to profile your use case before assuming that [`model_construct()`][pydantic.main.BaseModel.model_construct] is faster.
+    In Pydantic V2, the performance gap between validation (either with direct instantiation or the `model_validate*` methods)
+    and [`model_construct()`][pydantic.main.BaseModel.model_construct] has been narrowed
+    considerably. For simple models, going with validation may even be faster. If you are using [`model_construct()`][pydantic.main.BaseModel.model_construct]
+    for performance reasons, you may want to profile your use case before assuming it is actually faster.
 
-!!! warning
-    [`model_construct()`][pydantic.main.BaseModel.model_construct] does not do any validation, meaning it can create models which are invalid. **You should only
-    ever use the [`model_construct()`][pydantic.main.BaseModel.model_construct] method with data which has already been validated, or that you definitely trust.**
-
-```py
```

Comment:
This section was removed as it seems pretty specific. I updated the argument docstring of `_fields_set` to better reflect what it does.

Reply:
Agreed, this definitely doesn't need to be included - same request though, can we move this to `examples`?

Reply:
Imo the improved docstring is enough to understand what it does. Examples should probably cover "broader" aspects, such as integrating with 3rd party libraries (SQLAlchemy, logfire, etc) or how to implement complex core schema generation, etc.

Reply:
I suppose - let's link this to the `examples` issue just in case.